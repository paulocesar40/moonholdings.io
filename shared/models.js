export const nomicsLink = 'https://nomics.com/';
