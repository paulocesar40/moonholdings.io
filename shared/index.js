export { moonHoldings, nomicsLink } from './copy';
