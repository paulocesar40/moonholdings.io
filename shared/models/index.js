export { moonHoldings, nomicsLink } from './copy'
export { defaultSquareStyle } from './coins'
