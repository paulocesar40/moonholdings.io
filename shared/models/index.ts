export * from './assets'
export * from './copy'
export * from './squares'
