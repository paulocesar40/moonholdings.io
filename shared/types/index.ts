export * from './portfolio'
export * from './styles'
