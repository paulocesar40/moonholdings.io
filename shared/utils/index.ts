export * from './math'
export * from './modifiers'
