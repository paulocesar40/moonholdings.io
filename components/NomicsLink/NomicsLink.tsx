import React from 'react';
import styled from 'styled-components';

import { nomicsLink } from '../../shared/models';

export const Link = styled.p`
  position: absolute;
  z-index: 10;
  bottom: 0.4rem;
  right: 300px;
  width: 100%;
  font-size: 1rem;
  text-align: right;
  cursor: pointer;

  a {
    color: ${(props) => props.theme.apricot};
    cursor: pointer;

    :hover {
      color: ${(props) => props.theme.offWhite};
    }
  }
`;

export default class NomicsLink extends React.Component {
  render() {
    return (<Link>Powered by <a href={nomicsLink} target="blank">Nomics APIs.</a></Link>)
  }
}
