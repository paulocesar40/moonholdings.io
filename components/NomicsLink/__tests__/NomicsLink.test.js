import React from 'react'
import Head from 'next/head'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import NomicsLink, { Link } from '../NomicsLink'
import { nomicsLink } from '../../../shared'

describe('<NomicsLink /> component', () => {
  describe('rendering', () => {
    const wrapper = shallow(<NomicsLink />);
    it('should render a component matching the snapshot', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      expect(wrapper).toHaveLength(1);
      expect(wrapper.contains(
        <Link>Powered by <a href={nomicsLink} target="blank">Nomics APIs.</a></Link>
      )).toEqual(true);
    });
  });

  describe('The Nomics API link', () => {
    it('should be correct', () => {
      expect(nomicsLink).toEqual("https://nomics.com/");
    });
  });
});
