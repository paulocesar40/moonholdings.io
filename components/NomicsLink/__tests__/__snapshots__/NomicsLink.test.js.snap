// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NomicsLink /> component rendering should render a component matching the snapshot 1`] = `
<NomicsLink__Link>
  Powered by 
  <a
    href="https://nomics.com/"
    target="blank"
  >
    Nomics APIs.
  </a>
</NomicsLink__Link>
`;
